lib/sql.js

```diff
@@ -66,7 +66,13 @@ class SQL {
 
 		if (params) query.where(params);
 		if (limit) query.limit(limit);
-		if (offset) query.offset(offset);
+		if (offset) {
+			if (limit) {
+				query.offset(offset);
+			} else {
+				throw new Error('offset requires limit');
```

Comment:
Can you explain why offset requires limit?

Reply:
> OFFSET says to skip that many rows before beginning to return rows.
so if you don't have a limit then it's like offsetting first few x, then return all rows.

Reply:
In my opinion, even thougth offsetting first few x then return all rows are rare in real use case, the library may still have this functionality. Library should provide a complete functionality and let the user decide how to use it.